### PR TITLE
Knotx/knotx-fragments#92 demo scenario created

### PR DESCRIPTION
**Pull request by oskarjerzyk**
## Description
The purpose of this PR is to add better, more complex assertions to functional test validating the behavior of templating engines

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: src/functionalTest/java/io/knotx/stack/functional/KnotxFragmentsDebugDataWithHandlebarsTest.java [hunk 1]
```diff
@@ -94,6 +97,15 @@ void requestPage(VertxTestContext testContext, Vertx vertx,
...
             assertNotNull(nodeLog);
             assertEquals(8, nodeLog.getJsonObject("log").getJsonArray("operations").size());
+            JsonArray nodes = nodeLog.getJsonObject("log").getJsonArray("operations");
+            assertEquals("fetch-user-info", nodes.getJsonObject(0).getString("node"));
+            assertEquals("fetch-payment-providers", nodes.getJsonObject(1).getString("node"));
+            assertEquals("fetch-offers", nodes.getJsonObject(2).getString("node"));
+            assertEquals("fetch-offers-fallback", nodes.getJsonObject(3).getString("node"));
+            assertEquals("fetch-delivery-options-cb", nodes.getJsonObject(4).getString("node"));
+            assertEquals("fetch-delivery-timeout", nodes.getJsonObject(5).getString("node"));
+            assertEquals("composite", nodes.getJsonObject(6).getString("node"));
```
Comment (tomaszmichalak) [inline src/functionalTest/java/io/knotx/stack/functional/KnotxFragmentsDebugDataWithHandlebarsTest.java]: Composite nodes entries can be different. See the example below:
- composite `[A -> B, C -> D]`
- possible outputs: [A, B, C, D], [C, D, A, B]
Please fix the validation rules - check `io.knotx.fragments.engine.TaskEngineCompositeNodeTest#expectSuccessEventLogEntry`.

Comment (oskarjerzyk) [reply src/functionalTest/java/io/knotx/stack/functional/KnotxFragmentsDebugDataWithHandlebarsTest.java]: Unfortunately, there's no possibility to get a more detailed message regarding this entry

File: src/functionalTest/java/io/knotx/stack/functional/KnotxFragmentsDebugDataWithHandlebarsTest.java [hunk 2]
```diff
@@ -94,6 +100,24 @@ void requestPage(VertxTestContext testContext, Vertx vertx,
...
+            assertEquals("fetch-user-info", nodes.getJsonObject(0).getString("node"));
+
+            List<String> actions = new ArrayList<>(); //all actions which will occur in the composite node execution
+            actions.add("fetch-payment-providers");
+            actions.add("fetch-offers");
+            actions.add("fetch-offers-fallback");
+            actions.add("fetch-delivery-options-cb");
+            actions.add("fetch-delivery-timeout");
+
+            for (int i = 1; i < 6; i++) {
```
Comment (tomaszmichalak) [inline src/functionalTest/java/io/knotx/stack/functional/KnotxFragmentsDebugDataWithHandlebarsTest.java]: Can you please make this more verbose:
`assertEqualsWithRange("fetch-payment-providers", nodes, 1, 6)`

File: src/functionalTest/java/io/knotx/stack/functional/KnotxFragmentsDebugDataWithHandlebarsTest.java [hunk 3]
```diff
@@ -94,10 +100,25 @@ void requestPage(VertxTestContext testContext, Vertx vertx,
...
           }
 
           //only 2 fragments, matcher should fail looking for next entries
           assertFalse(matcher.find());
         });
   }
+
+  private void assertEqualsWithinRange(String object, JsonArray collection, int lowerBound,
+      int upperBound) {
+    List<LinkedHashMap> entries = collection.getList().subList(lowerBound, upperBound);
```
Comment (tomaszmichalak) [inline src/functionalTest/java/io/knotx/stack/functional/KnotxFragmentsDebugDataWithHandlebarsTest.java]: Much much better!

**Pull request merged**